### PR TITLE
Added CES Changes and Modified MROT playbook

### PR DESCRIPTION
**Pull request by jayeshh123**
Changes related to CES and Modified MROT playbook

### CODE REVIEW

**Comment by digvijay-ukirde** [comment]
Looks good.
@rajan-mis @mamuthiah Can you review the code please.

File: roles/auth_configure/tasks/user_auth_userdefined.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+# Task to setup the Auth
+
+- name: Setup Auth
+  shell: "/usr/lpp/mmfs/bin/mmuserauth service create --data-access-method file --type userdefined"
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true and service_result.rc != 0
```
Comment (rajan-mis) [inline roles/auth_configure/tasks/user_auth_userdefined.yml:7]: I don't think we need to have `is_admin_node`  flag in the when condition. you can run this from one of the protocol node .

File: roles/auth_prepare/tasks/auth_prepare.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+# Task to check if Auth is already setup
+
+- name: Check service is already exists
+  shell: "/usr/lpp/mmfs/bin/mmuserauth service check | grep -q 'File configuration is *'"
+  register: service_result
+  ignore_errors: yes
+  failed_when: service_result.rc == 2
+  when: is_admin_node | default(false) == true
```
Comment (rajan-mis) [inline roles/auth_prepare/tasks/auth_prepare.yml:9]: same is_admin_node not required , we should run these task from one of the CES node

File: roles/core_configure/tasks/storage_fileset.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true
+
+- name: Debug the filesets in file system
+  debug:
+    var: filesets.stdout_lines
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true
+
+# Setting up quota for filesets
```
Comment (rajan-mis) [inline roles/core_configure/tasks/storage_fileset.yml:68]: But i think Quota may not be applicable for all different set of deployment ,so it should be handled within some different boolean flag if user wants to set quota they can say , example: `scale_quota_enabled = true`

**Review by rajan-mis — COMMENTED** [review]
I am still  reviewing other parts but please check my some of the review comment. i don't think is_admin_node , we need to use for authentication and filesets .

File: roles/auth_configure/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+
+# tasks file for failover
+- import_tasks: user_auth_userdefined.yml
+  when: scale_protocols is defined and (scale_protocols.nfs|bool or scale_protocols.smb|bool) #and ldap_basedns == null
```
[the remark this answers is not in the document]
Comment (jayeshh123) [reply roles/auth_configure/tasks/main.yml]: Resolved

File: roles/auth_configure/tasks/user_auth_userdefined.yml (hunk 1 above)
Comment (jayeshh123) [reply roles/auth_configure/tasks/user_auth_userdefined.yml]: Resolved

File: roles/auth_prepare/tasks/auth_prepare.yml (hunk 2 above)
Comment (jayeshh123) [reply roles/auth_prepare/tasks/auth_prepare.yml]: Resolved

File: roles/core_configure/tasks/storage_fileset.yml (hunk 3 above)
Comment (jayeshh123) [reply roles/core_configure/tasks/storage_fileset.yml:64]: Resolved

File: roles/nfs_file_share/tasks/nfs_file_share.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
+# Checking if NFS export list if already exists if not then Creating NFS export.
+
+- name: Check NFS export list if already exists
+  shell: "/usr/lpp/mmfs/bin/mmnfs export list -Y | tail -n +2 | awk -F: '{print $7, $8}' | sort -k2 | awk '{print $1}' | grep -w {{ scale_protocols.mountpoint }}/{{ item }}"
+  loop: "{{ scale_protocols.exports }}"
+  register: export_list_exists
+  failed_when: export_list_exists.rc == 2
+  when: is_admin_node | default(false) == true
```
Comment (rajan-mis) [inline roles/nfs_file_share/tasks/nfs_file_share.yml:9]: Do we need is_admin_node for these task.

File: samples/vars/storage_fileset_vars.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+# Important:
+# `scale_storage` *must* be define using group variables. Do *not* define disk
+# parameters using host variables or inline variables in your playbook. Doing so
+# would apply them to all hosts in the group/play, thus defining the same disk
+# multiple times...
+
+scale_storage:
+  - filesystem: gpfs01
+    filsets:
```
Comment (rajan-mis) [inline samples/vars/storage_fileset_vars.yml:19]: Can you please confirm on the fileset quota size and other value that we were discussing

Comment (rajan-mis) [reply samples/vars/storage_fileset_vars.yml:19]: Does filsets: only contain name.  and looks variable name has typo issue

Comment (jayeshh123) [reply samples/vars/storage_fileset_vars.yml]: Resolved

https://github.com/jayeshh123/ibm-spectrum-scale-install-infra/blob/1504d87ec031d4b1194c3d51b63b530c41841edc/samples/vars/storage_fileset_vars.yml#L19

File: samples/vars/storage_fileset_vars.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+# Important:
+# `scale_storage` *must* be define using group variables. Do *not* define disk
+# parameters using host variables or inline variables in your playbook. Doing so
+# would apply them to all hosts in the group/play, thus defining the same disk
+# multiple times...
+
+scale_storage:
+  - filesystem: gpfs01
+    filesets:
```
Comment (rajan-mis) [inline samples/vars/storage_fileset_vars.yml:19]: Can you please also mention, how user can specify fileset names. i think this should also consider multiple FSETname.

**Review by rajan-mis — APPROVED** [review]
LGM! Please open an issue for discussed improvement. Thanks

**Pull request merged**